### PR TITLE
Wave default values and smart logic

### LINKED ISSUE #20: Add support for "smarter" enemies that can cast abilities
Currently, all waves have a "dumb" AI. Generally, all the AI knows to do is try to touch you, dying in the process. However, enemies should have an improved AI that allows it "dumb casting" of abilities, where abilities are cast whenever they're off cooldown. Abilities that can affect the player will always be directed towards it, in its direction or position. Abilities that are no target are simply cast, and abilities that affects allies will always be directed towards the nearest living ally.

Bosses might have a smart AI implemented specifically for each boss to allow them to be smarter than regular waves.

### PR DESCRIPTION
**Pull request by Shushishtok**
Fixes #20 

- Localized existing wave names.
- Added default values for: unit health, unit damage, unit move speed, wave interval and wave max unit count. Those can now be omitted when defining a wave, or overridden if we want a specific value to be used.
- Added four AI logic types. They're designated to stay dumb, but control the way the AI controls its units. The options are:
    - Touch only: the unit simply try to touch the player, dying and dealing damage when it does.
    - Cast only: the unit only casts whatever abilities it has, otherwise it is idle. Useful for units that has some kind of projectile that they will try to fire at the player with very low cooldown.
    - Cast and touch: the unit tries to casts whatever abilities it has. If it cannot, it will try to touch the player instead. Useful for units that have abilities but should still try to touch the player.
    - Run away: currently unique to round 1. Not fully implemented yet. Units with this AI logic will constantly try to run away from the player.
    
**Note:** ability casting AI is currently "dumb" - it assumes a single target unit targeting ability on the player. Later on, as we design and implement more abilities, we can rework the AI to take multiple ability behavior types and handle them accordingly.

- Added direction calculation util functions.
- Added util functions for determining whether a unit is capable of casting an ability on a target, if any. (target is optional, e.g. on cases of No Target abilities).
- Localization files can now reference types/enums found on the d.ts. files in the /common directory.
- Renamed Sides to Side.

### CODE REVIEW

File: game/scripts/vscripts/core/wave_manager.ts [hunk 1]
```diff
@@ -318,6 +335,43 @@ class UnitAI extends BaseModifier {
...
+				break;
+
+			case WaveUnitAILogicType.CAST_ONLY:
+				this.CastAbilities();
+				break;
+
+			case WaveUnitAILogicType.CAST_AND_TOUCH:
+				if (!this.CastAbilities()) {
+					this.MoveToTouchPlayer();
+				}
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/wave_manager.ts:350]: missing break here

**Review by Nibuja05 — CHANGES_REQUESTED** [review]
Good job! Looks good so far
Can you also take a look at my changes? 😅

File: game/scripts/vscripts/core/wave_manager.ts (hunk 1 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/wave_manager.ts:350]: Fixed in latest commit

**Review by Nibuja05 — APPROVED** [review]
Nice, everythings ok now 👍

**Pull request merged**